### ISSUE TITLE
Check containers are super ambiguous in `fly containers`

### ISSUE THREAD

**Issue by clarafu** [concourse/concourse#1149]
Hello!

# Feature Request

## What challenge are you facing?

In **2.8.0**, `fly containers` for check containers give almost no useful information revolving around which resource is being checked or what the check container is being used for. The new check container doesn't conform to the current `fly containers` format (for example, they are not associated to a pipeline, job, build, resource name). 

```
$ fly -t ci containers
handle                                worker                                pipeline      job                build #  build id  type   name                         attempt
003680d4-d0eb-4c8f-4572-ecc0c9c5f77b  40e8b592-7f58-449c-9c68-e43edd57d778  images        build-buildbox-ci  11       35592     put    concourse-buildbox-ci-image  n/a
00daa352-ab45-4c80-696e-779c5df3654f  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
02f25799-4a0f-490e-6495-9c07c8f22068  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
030143d3-cafb-4d48-562f-a7c1f9482633  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
05406600-42ca-4ef7-5b05-5ed7ca2015e1  windows-worker                        none          none               none     none             none                         n/a
0697f6c7-e43c-4245-4d11-a2022bb55886  windows-worker                        none          none               none     none             none                         n/a
086958d5-6fc3-4c2c-5f9b-aaf639510b9e  cfee4482-405c-4fa7-b7cf-256e69c7af92  none          none               none     none      check  none                         n/a
12b597bd-1be2-43b6-42eb-6db97d2c0d67  4fd44d6a-b86c-41f1-835d-73968ad37298  none          none               none     none      check  none                         n/a
12ed1977-052b-4503-6c45-6f3ff12edc4d  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
13542901-7d1b-4ae9-6f84-a016c65d3a5e  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
1551a9ec-d311-4ed9-530f-aae01cf17a0c  4fd44d6a-b86c-41f1-835d-73968ad37298  none          none               none     none      check  none                         n/a
163f2b62-afeb-4a12-4d68-4220781c119a  cfee4482-405c-4fa7-b7cf-256e69c7af92  none          none               none     none      check  none                         n/a
1a7b8726-4d4c-41cd-768e-0800a055895c  40e8b592-7f58-449c-9c68-e43edd57d778  none          none               none     none      check  none                         n/a
1b11da52-934b-4307-6808-53e7ae7d92bf  4fd44d6a-b86c-41f1-835d-73968ad37298  none          none               none     none      check  none                         n/a
1fd85113-6a37-4e65-5a4c-384d9be1113a  4fd44d6a-b86c-41f1-835d-73968ad37298  none          none               none     none      check  none                         n/a
20bed82f-f587-45a8-5d3b-9f22719cba96  40e8b592-7f58-449c-9c68-e43edd57d778  prs           atc                69       35740     put    atc                          n/a
```

## A Modest Proposal

A solution we could take is to create a new fly command, maybe `fly resources`, that lists the check containers and some details about the corresponding resource config (type, id?, but not the resource config because it might contain credentials). It could also be useful to have columns that unravel the uses of the config in pipelines, resources, etc..

**Comment by geofffranks** [comment]
+1

**Comment by DRuggeri** [comment]
+1
This would be very helpful for those hitting the "max containers" errors

**Comment by jyothi-eroader** [comment]
+1 
Will be really useful when investigating "Insufficient subnets in pool" when we max out at 256 containers

**Comment by edtan** [comment]
I've added some PRs to get started on a new `fly resources` command.  This was largely based off of the existing `fly containers` command.

There are two 'modes' - the default mode gives you a summary of all the check containers which should match up 1 to 1 with the check containers listed in `fly containers`.  If you pass in `--detailed`, it gives you more information (e.g. pipelines and resource names) that the check containers are used in.  Because there's a single check container per resource config, but resource configs can be used in more than 1 pipeline, `--detailed` will usually return duplicate rows for a resource config corresponding to the different pipelines and resource names.

**Comment by marco-m** [comment]
@edtan thanks for the PRs! I think it would be useful for the reviewers if you added in the PR for fly some sample outputs of your command (disclaimer: I am just a happy Concourse user).

**Comment by edtan** [comment]
@marco-m Good idea!  I've updated the fly PR with sample output of the command.

**Comment by wagdav** [comment]
@clarafu  Given the argument https://github.com/concourse/concourse/issues/1374#issuecomment-462368187 and the PRs from @edtan  I think this issue can be closed in favor of #3079

**Comment by vito** [comment]
yup
